examples/Microsoft.Spark.CSharp.Examples/MachineLearning/Sentiment/README.md

```diff
@@ -0,0 +1,188 @@
+
+### 2. Build and Train Your Model
+
+Follow the [Model Builder Getting Started Guide](https://dotnet.microsoft.com/learn/machinelearning-ai/ml-dotnet-get-started-tutorial/intro) to train your model using the sentiment analysis scenario. 
+
+Follow the steps to:
+
+* Create a new C# Console App
+* Pick the **Sentiment Analysis** scenario
+* Train using the **Amazon.csv** dataset
+
+![Sentiment Analysis Model Builder](https://dotnet.microsoft.com/static/images/model-builder-vs.png?v=9On8qwmGIXdAyX_-zAmATwYU7fd7tzem-_ojnv1G7XI)
+
+### 3. Generate Model and Code
+
+In the last step of using Model Builder, you'll produce a zip file containing the trained ML.NET model. In the image below, you can see it's contained in **MLModel.zip.**
+
+![ML.NET Zip and Files](https://github.com/bamurtaugh/spark/blob/SparkMLNet/examples/Microsoft.Spark.CSharp.Examples/MachineLearning/images/modelbuilder5proj.PNG)
+
+You'll also generate C# code you can use to consume your model in other .NET apps, like the Spark app we'll be creating.
+
+![Generated Code](https://github.com/bamurtaugh/spark/blob/SparkMLNet/examples/Microsoft.Spark.CSharp.Examples/MachineLearning/images/modelbuilder5code.PNG)
+
+### 4. Add ML.NET to .NET for Apache Spark App
+
+You have a few options to start creating a .NET for Apache Spark app that uses this ML.NET code and trained model. Make sure that in any app you develop, you've downloaded the [Microsoft.ML NuGet Package](https://www.nuget.org/packages/Microsoft.ML).
+
+#### Option 1: Add Projects
+
+One option is to use Model Builder's *Add Projects* feature, which will result in 3 projects:
+
+* Your original app (**myMLApp**)
+* A console app that allows you to build/train/test the model (**myMLAppML.ConsoleApp**)
+* A .NET Standard class library that contains model input/output and your trained model in a zip file (**myMLAppML.Model**)
+
+You would begin writing your .NET for Apache Spark code (and paste in the code generated from Model Builder, shown in step 3 above) in the original app **myMLApp.**
+
+![Model Builder Result](https://dotnet.microsoft.com/static/images/model-builder-generated-code.png?v=iC-r8k3zpKUwQVoNOH34D903IhXhIb4CsX003484s7c)
+
+#### Option 2: Create a new console app (shown in this repo)
+
+Rather than working with the projects/files produced by Model Builder's Add Projects, you can create a new, separate C# console app. You just need to copy over your model's zip file to a directory your new console app can access. In this repo, a trained model **MLModel.zip** has already been included for you in the *Resources* folder.
+
+As we create the logic for our Spark app, we'll paste in the code generated from Model Builder and include some other class definitions.
+
+## .NET for Spark
+
+Now that we've trained an ML.NET model for sentiment analysis, we can begin writing the .NET for Spark code that will read in our Yelp data, pass each review to our ML.NET model, and predict whether reviews are positive or negative.
```

Comment:
Just a general ML question.  Is it okay to mix different types of reviews like this ?  The MLModel trained using the Amazon dataset, which are reviews on the quality of a product.   Yelp reviews would be reviews on food/decor/service.

Reply:
I believe it should be acceptable in this use case. Both datasets seem general enough that the model can learn what a user says when they're positive vs when they're negative about something (whether it's product quality or food/service). 

i.e. it'll catch that terms like "great" "very good" and "ideal" are positive, while "unacceptable" "waste" and "unacceptable" are negative, and both datasets contain terms like these.

Reply:
Also, I found these datasets through the [ML.NET sentiment analysis tutorial](https://docs.microsoft.com/en-us/dotnet/machine-learning/tutorials/sentiment-analysis), so I also took that as a sign that it should be fine.

Reply:
Generally in the ML world if there is a small dataset, then they split it into a training set and a test set.   From the link you gave, they only use the yelp dataset.  The ML library looks it reads the yelp dataset and will createa two subsets, a training set (to create the model) and a test set (to do testing and validations).

Reply:
Yep I see where they have divided Yelp dataset into training and testing. Do you think trying to integrate something like that would be preferable/not too confusing rather than using the 2 separate datasets as the sample stands now?

Reply:
I've done a little dabbling with ML, but my only concern is that I'm just unsure how legitimate this example would seem to an ML expert.

Reply:
Yeah I understand what you're saying. I feel like with sentiment analysis it could be okay since we're just focusing on simple text, but to improve accuracy/legitimacy, I'm going to divide the Yelp dataset in half and train with one half of it and test with the other half.

Reply:
Just updated, thanks for the feedback